Pharmacy/PharamcyApi/Controllers/MedicineController.cs

```diff
@@ -26,13 +28,23 @@ public Medicine Add(Medicine medicine)
         [HttpGet]
         public IEnumerable<Medicine> GetAll()
         {
-            return _uow.GetRepository<IMedicineReadRepository>().GetAll();
+            return _uow.GetRepository<IMedicineReadRepository>().GetAll().Include(medicine => medicine.Manufacturer);
         }
 
         [HttpGet("{id}")]
         public Medicine GetById(int id)
         {
-            return _uow.GetRepository<IMedicineReadRepository>().GetById(id);
+            return _uow.GetRepository<IMedicineReadRepository>().GetAll()
+                .Include(medicine => medicine.Manufacturer)
+                .FirstOrDefault(medicine => medicine.Id == id);
+        }
+
+        [HttpGet("{name}")]
+        public Medicine GetByName(string name)
+        {
+            return _uow.GetRepository<IMedicineReadRepository>().GetAll()
```

Comment:
Would check if we have it in the db and if we do return it, but if left like this even if we don't have it the request returns HttpStatusCode.OK rather than HttpStatusCode.BadRequest

Reply:
404 NotFound would be ideal in my opinion, since the request is valid, it's just that entity doesn't exist in db